examples/passport-login/src/authentication-strategies/google.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: example-passport-oauth2-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  asAuthStrategy,
+  AuthenticationStrategy,
+  UserIdentityService,
+  AuthenticationBindings,
+} from '@loopback/authentication';
+import {StrategyAdapter} from '@loopback/authentication-passport';
+import {Profile} from 'passport';
+import {Strategy, StrategyOptions} from 'passport-google-oauth2';
+import {bind, inject} from '@loopback/context';
+import {UserServiceBindings} from '../services';
+import {extensionFor} from '@loopback/core';
+import {securityId, UserProfile} from '@loopback/security';
+import {User} from '../models';
+import {Request, RedirectRoute} from '@loopback/rest';
+import {PassportAuthenticationBindings} from './oauth2';
+
+@bind(
+  asAuthStrategy,
+  extensionFor(PassportAuthenticationBindings.OAUTH2_STRATEGY),
+)
+export class GoogleOauth2Authorization implements AuthenticationStrategy {
+  name = 'oauth2-google';
+  passportstrategy: Strategy;
+  protected strategy: StrategyAdapter<User>;
+
+  /**
+   * create an oauth2 strategy for google
+   */
+  constructor(
+    @inject(UserServiceBindings.PASSPORT_USER_IDENTITY_SERVICE)
+    public userService: UserIdentityService<Profile, User>,
+    @inject('googleOAuth2Options')
+    public googleOptions: StrategyOptions,
+  ) {
+    this.passportstrategy = new Strategy(googleOptions, this.verify.bind(this));
+    this.strategy = new StrategyAdapter(
```

Comment:
This setup for google doesn't match what you have for FaceBook or Local
where you use .bind(this)

```
    this.passportstrategy = new Strategy(
      facebookOptions,
      this.verify.bind(this),
    );
    this.strategy = new StrategyAdapter(
      this.passportstrategy,
      this.name,
      this.mapProfile.bind(this),
    );
```

Reply:
agreed, I will fix this.